acceptance/tests/test_upgrade_puppet6_to_puppet7_yum.rb

```diff
@@ -0,0 +1,43 @@
+require 'beaker-puppet'
+require_relative '../helpers'
+
+# Tests FOSS upgrades from the latest puppet 6 (the puppet6 collection) to puppet 7.0.0.
+test_name 'puppet_agent class: Upgrade agents from puppet6 to puppet7' do
+  confine :to, :platform => /el-|fedora-/
+  # require_master_collection 'puppet7' #once puppet7 collection is available this can be uncommented
+  
+  exclude_pe_upgrade_platforms
+  puppet_testing_environment = new_puppet_testing_environment
+  step "Create new site.pp with upgrade manifest" do
+    manifest = <<-PP
+    node default {
+      class { puppet_agent:
+        package_version => '7.0.0.224.gaf2034af',
```

Comment:
Maybe we should upgrade to `latest` as nightly versions change quite often(and a cleanup is done for old versions) and we need to ensure that we can upgrade.

Reply:
I think puppetlabs-puppet_agent does not support `latest` in its manifest. It can support `auto` but it will only update to the master server version.
Once puppet7 is released we won't have to specify a source for the package, but until then I can't think of a better workaround.

Reply:
The issue with fix version like `7.0.0.224.gaf2034af` is that old versions get removed from nightlies, so we might end-up with this test failing because that version is not found.

Also the test is confined only on debian based platforms, which is a bit restrictive.

If we want just to have a sanity check that the upgrade just work on some platforms, we can use osx/windows which are not relying on a package provider. There we have `puppet-agent-latest` packages.

You can also take a look on how the bolt task is testing the upgrade from puppet 6 to puppet 7 nightly,

Reply:
:+1:, maybe we can programatically get the latest build available in a nightly collection. i.e. by `curl`-ing http://builds.delivery.puppetlabs.net/passing-agent-SHAs/puppet-agent-main-version and replacing `6.18.0` with `7.0.0`